scripts/gen.py

```diff
@@ -880,7 +889,8 @@ def extract_functions(path):
     fndef = m.group(1)
     try:
       _XPARSER.parse(fndef)
-      functions.append(FuncDef(cpp_sig=fndef, aten_sig=m.group(2)))
+      comment = json.loads(m.group(2))
+      functions.append(FuncDef(cpp_sig=fndef, aten_sig=comment['schema'], compound=comment['compound']=='1'))
```

Comment:
I would make an API:
```Python
def _create_funcdef(fndef, jdata):
  ...
  return FuncDef(cpp_sig=fndef, ...)
```
That creates a `FuncDef` out of the `fndef` and JSON data.